src/Core/Cache/ManifestCacheFactory.php

```diff
@@ -70,23 +79,22 @@ public function create($service, array $params = [])
     }
 
     /**
-     * Create cache directly without config / injector
-     *
-     * @param string $class
-     * @param array $args
-     * @return CacheInterface
+     * Convert class names of PSR-16 implementaions that were in symfony 4.x until removed in symfony 5.x
+     * and use their PSR-6 replacements
      */
-    public function createCache($class, $args)
+    private function convertLegacySymfonyClassName(string $cacheClass): string
     {
-        /** @var CacheInterface $cache */
-        $reflection = new ReflectionClass($class);
-        $cache = $reflection->newInstanceArgs($args);
-
-        // Assign cache logger
-        if ($this->logger && $cache instanceof LoggerAwareInterface) {
-            $cache->setLogger($this->logger);
+        $map = [
+            'PhpFilesCache' => PhpFilesAdapter::class,
+            'FilesystemCache' => FilesystemAdapter::class,
+            'ApcuCache' => ApcuAdapter::class,
+            'ChainCache' => ChainAdapter::class,
+        ];
+        foreach ($map as $match => $newClass) {
+            if ($cacheClass === "Symfony\\Component\\Cache\\Simple\\$match") {
+                return $newClass;
+            }
```

Comment:
This feels like something that would be needed if we were doing this in 4.x - why do we need to do this here? Shouldn't we just update any references we have to the old class names?
If developers need to update their own environment variables then so be it - that's what the upgrade guide (or changelog) is for.

If we _do_ keep this in, we should include a deprecation notice and remove this in CMS 6.... but it feels a bit of an anti-pattern to _start_ CMS 5 with new deprecations.

Reply:
The assumption here is that developers are able to able to easily update environment variables - based on [this comment](https://github.com/silverstripe/silverstripe-framework/issues/10345#issuecomment-1170683316) from the community that assumption doesn't always hold.

You could can of course make the argument that major upgrades have an expectation of just being harder.

I'm 50/50 on this one - what's your preference here?

Reply:
I'd prefer to not include this mapping. The fact that this is a major release does mean that there will be breaking changes, and sometimes that might mean making changes to environment variables. For minors I could see it being a problem, but majors come once every two years so I think people can live with it.

Reply:
OK, will remove